### PR TITLE
Fix filter order reset

### PR DESCRIPTION
**Pull request by pdamasceno**
We were not updating the order in the filters container in the cache manager
The first time we get the filters from a source works fine because we get them directly from libobs (sdi->filtersOrderChanged = true)
Subsequent calls to get filters start getting them from the cache manager and there the order was not right

### CODE REVIEW

File: obs-studio-client/source/input.hpp [hunk 1]
```diff
@@ -26,6 +26,14 @@ namespace osn
 {
 	class Input : public osn::ISource, public utilv8::ManagedObject<osn::Input>
 	{
+		enum FilterMovement
```
Comment (summeroff) [inline obs-studio-client/source/input.hpp]: int values of this enums have fixed values on frontend. 
So better to have them explicitly set in this enum too and not expect that compiler always generate them exactly as we need.

Comment (pdamasceno) [reply obs-studio-client/source/input.hpp]: Ok! Making the change

File: obs-studio-client/source/input.cpp [hunk 2]
```diff
@@ -908,6 +910,30 @@ Nan::NAN_METHOD_RETURN_TYPE osn::Input::SetFilterOrder(Nan::NAN_METHOD_ARGS_TYPE
 
 	SourceDataInfo* sdi = CacheManager<SourceDataInfo*>::getInstance().Retrieve(baseobj->sourceId);
 	if (sdi) {
+		// Update order on cache manager filter container
+		auto it = std::find(sdi->filters->begin(), sdi->filters->end(), basefilter->sourceId);
+
+		if (movement == FilterMovement::UP && it != sdi->filters->begin()) {
+			std::iter_swap(it, it - 1);
+		}
```
Comment (summeroff) [inline obs-studio-client/source/input.cpp]: This 4 if's can be joined by "else" or replaced by "switch" as they expected to work exclusively. If one of them had worked no need to check another.

Comment (pdamasceno) [reply obs-studio-client/source/input.cpp]: You are right. I'll change it to a switch

**Review by summeroff — APPROVED** [review]

**Review by summeroff — APPROVED** [review]

**Review by EddyGharbi — APPROVED** [review]

**Review by summeroff — APPROVED** [review]

**Pull request merged**